GoogleHadoopFileSystemIntegrationHadoop2Test fails in V1.9.17

I am trying to use this feature. https://github.com/GoogleCloudDataproc/hadoop-connectors/commit/02b17d1b11369e53a75f48f2ce9e064e040070d9

This contains GoogleHadoopFileSystemIntegrationHadoop2Test , so I tried running it but it fails saying account not valid even though I am able to use my account for listing and updates. 

I can see that this test has been removed from latest version. Could somebody help me running this test as well as make me understand why this class has been removed. Also if this class has been removed, where are the tests for feature "Add Hadoop File System extended attributes support."

Thanks.

Yesterday I cherry picked just the above mentioned commit https://github.com/GoogleCloudDataproc/hadoop-connectors/commit/02b17d1b11369e53a75f48f2ce9e064e040070d9  in my older v1.9.10 and was able to setup the same test GoogleHadoopFileSystemIntegrationHadoop2Test using my Intellij as well. 
This is the full stacktrace I get.

`Caused by: com.google.api.client.auth.oauth2.TokenResponseException: 400 Bad Request
{
  "error" : "invalid_grant",
  "error_description" : "Invalid grant: account not found"
}
	at com.google.api.client.auth.oauth2.TokenResponseException.from(TokenResponseException.java:105)
	at com.google.api.client.auth.oauth2.TokenRequest.executeUnparsed(TokenRequest.java:287)
	at com.google.api.client.auth.oauth2.TokenRequest.execute(TokenRequest.java:307)
	at com.google.cloud.hadoop.util.CredentialFactory$GoogleCredentialWithRetry.executeRefreshToken(CredentialFactory.java:162)
	at com.google.api.client.auth.oauth2.Credential.refreshToken(Credential.java:493)
	at com.google.api.client.auth.oauth2.Credential.intercept(Credential.java:217)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:868)`

--- a/tools/run_integration_tests.sh
+++ b/tools/run_integration_tests.sh
@@ -50,19 +50,30 @@ check_required_params() {
   fi
 }
 
-export HDFS_ROOT=file:///tmp
-
 check_required_params
 
 # When tests run, they run in the root of the module directory. Anything
 # relative to our current directory won't work properly
 if [[ $(uname -s) == Darwin ]]; then
-  # On MacOS `readlink` does not support `-e` parameter
-  GCS_TEST_PRIVATE_KEYFILE=$(readlink -n "${GCS_TEST_PRIVATE_KEYFILE}")
+  # On MacOS `readlink` does not support `-f` parameter
+  # so we need manually resolve absolute path and symlink after that
+  abs_path() {    
+   echo "$(cd $(dirname "$1"); pwd)/$(basename "$1")"
+  }
+  GCS_TEST_PRIVATE_KEYFILE=$(abs_path "$GCS_TEST_PRIVATE_KEYFILE")
+  while [[ -L $GCS_TEST_PRIVATE_KEYFILE ]]; do
+    GCS_TEST_PRIVATE_KEYFILE=$(readlink -n "$GCS_TEST_PRIVATE_KEYFILE")
+  done
 else
-  GCS_TEST_PRIVATE_KEYFILE=$(readlink -n -e "${GCS_TEST_PRIVATE_KEYFILE}")
+  GCS_TEST_PRIVATE_KEYFILE=$(readlink -n -f "${GCS_TEST_PRIVATE_KEYFILE}")
 fi
+if [[ ! -f $GCS_TEST_PRIVATE_KEYFILE ]]; then
+  echo "Resolved '${GCS_TEST_PRIVATE_KEYFILE}' keyfile does not exist."
+  exit 1
+fi
+
 export GCS_TEST_PRIVATE_KEYFILE
+export HDFS_ROOT=file:///tmp
 export RUN_INTEGRATION_TESTS=true
 
 ./mvnw -B -e -T1C "-P${HADOOP_VERSION}" -Pintegration-test clean verify "${@:5}"

--- a/tools/run_integration_tests.sh
+++ b/tools/run_integration_tests.sh
@@ -60,9 +60,9 @@ if [[ $(uname -s) == Darwin ]]; then
   abs_path() {    
    echo "$(cd $(dirname "$1"); pwd)/$(basename "$1")"
   }
-  GCS_TEST_PRIVATE_KEYFILE=$(abs_path "$GCS_TEST_PRIVATE_KEYFILE")
+  GCS_TEST_PRIVATE_KEYFILE=$(abs_path "${GCS_TEST_PRIVATE_KEYFILE}")
   while [[ -L $GCS_TEST_PRIVATE_KEYFILE ]]; do
-    GCS_TEST_PRIVATE_KEYFILE=$(readlink -n "$GCS_TEST_PRIVATE_KEYFILE")
+    GCS_TEST_PRIVATE_KEYFILE=$(readlink -n "${GCS_TEST_PRIVATE_KEYFILE}")
   done
 else
   GCS_TEST_PRIVATE_KEYFILE=$(readlink -n -f "${GCS_TEST_PRIVATE_KEYFILE}")

--- a/tools/run_integration_tests.sh
+++ b/tools/run_integration_tests.sh
@@ -60,7 +60,7 @@ if [[ $(uname -s) == Darwin ]]; then
   # On MacOS `readlink` does not support `-e` parameter
   GCS_TEST_PRIVATE_KEYFILE=$(readlink -n "${GCS_TEST_PRIVATE_KEYFILE}")
 else
-  GCS_TEST_PRIVATE_KEYFILE=$(readlink -e "${GCS_TEST_PRIVATE_KEYFILE}")
+  GCS_TEST_PRIVATE_KEYFILE=$(readlink -n -e "${GCS_TEST_PRIVATE_KEYFILE}")
 fi
 export GCS_TEST_PRIVATE_KEYFILE
 export RUN_INTEGRATION_TESTS=true

--- a/tools/run_integration_tests.sh
+++ b/tools/run_integration_tests.sh
@@ -56,7 +56,12 @@ check_required_params
 
 # When tests run, they run in the root of the module directory. Anything
 # relative to our current directory won't work properly
-GCS_TEST_PRIVATE_KEYFILE=$(readlink -f "${GCS_TEST_PRIVATE_KEYFILE}")
+if [[ $(uname -s) == Darwin ]]; then
+  # On MacOS `readlink` does not support `-e` parameter
+  GCS_TEST_PRIVATE_KEYFILE=$(readlink -n "${GCS_TEST_PRIVATE_KEYFILE}")
+else
+  GCS_TEST_PRIVATE_KEYFILE=$(readlink -e "${GCS_TEST_PRIVATE_KEYFILE}")
+fi
 export GCS_TEST_PRIVATE_KEYFILE
 export RUN_INTEGRATION_TESTS=true
 

